java-spiffe-core/src/main/java/spiffe/bundle/jwtbundle/JwtBundle.java

```diff
@@ -0,0 +1,195 @@
+import spiffe.Algorithm;
+import spiffe.bundle.BundleSource;
+import spiffe.exception.AuthorityNotFoundException;
+import spiffe.exception.BundleNotFoundException;
+import spiffe.exception.JwtBundleException;
+import spiffe.spiffeid.TrustDomain;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.KeyException;
+import java.security.PublicKey;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A <code>JwtBundle</code> represents a collection of trusted JWT authorities (Public Keys) for a trust domain.
+ */
+@Value
+public class JwtBundle implements BundleSource<JwtBundle> {
+
+    TrustDomain trustDomain;
+
+    Map<String, PublicKey> jwtAuthorities;
+
+    /**
+     * Creates a new JWT bundle for a trust domain.
+     *
+     * @param trustDomain a {@link TrustDomain} to associate to the JwtBundle
+     */
+    public JwtBundle(@NonNull TrustDomain trustDomain) {
+        this.trustDomain = trustDomain;
+        this.jwtAuthorities = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Creates a new JWT bundle for a trust domain with JWT Authorities (public keys associated to keyIds).
+     *
+     * @param trustDomain    a {@link TrustDomain} to associate to the JwtBundle
+     * @param jwtAuthorities a Map of public Keys
+     */
+    public JwtBundle(@NonNull TrustDomain trustDomain, @NonNull Map<String, PublicKey> jwtAuthorities) {
+        this.trustDomain = trustDomain;
+        this.jwtAuthorities = new ConcurrentHashMap<>(jwtAuthorities);
+    }
+
+    /**
+     * Loads a JWT bundle from a file on disk. The file must contain a standard RFC 7517 JWKS document.
+     * <p>
+     * Key Types supported are EC and RSA.
+     *
+     * @param trustDomain a {@link TrustDomain} to associate to the JWT bundle.
+     * @param bundlePath  a path to a file containing the JWT authorities (public keys).
+     * @return a instance of a {@link JwtBundle}
+     * @throws JwtBundleException if there is an error reading or parsing the file, or if a keyId is empty
+     * @throws KeyException       if the bundle file contains a key type that is not supported
+     */
+    public static JwtBundle load(@NonNull final TrustDomain trustDomain, @NonNull final Path bundlePath) throws KeyException, JwtBundleException {
+        try {
+            val jwkSet = JWKSet.load(bundlePath.toFile());
+            return toJwtBundle(trustDomain, jwkSet);
+        } catch (IOException | ParseException | JOSEException e) {
+            throw new JwtBundleException(String.format("Could not load bundle from file: %s", bundlePath.toString()), e);
+        }
+    }
+
+    /**
+     * Parses a JWT bundle from a byte array.
+     *
+     * @param trustDomain a {@link TrustDomain}
+     * @param bundleBytes an array of bytes representing the JWT bundle.
+     * @return an instance of a {@link JwtBundle}
```

Comment:
Missing JavaDoc comments on when `KeyException` and `JwtBundleException` is thrown from this method

Reply:
Added javadoc comments.